lib/web_ui/lib/src/engine/canvaskit/embedded_views.dart

```diff
@@ -504,12 +504,12 @@ class HtmlViewEmbedder {
     }
 
     // Add all the pictures from the deleted canvases to the second-to-last
-    // canvas.
+    // canvas (or the last canvas if there is only one).
     sawLastCanvas = false;
```

Comment:
Another fix may be to do:

```
sawLastCanvas = (maximumCanvases == 1);
```

Instead of initializing it directly to `false`? That way you save an "or" on the for loop below.

Reply:
Done